openedx/core/djangoapps/geoinfo/middleware.py

```diff
@@ -33,14 +33,13 @@ def process_request(self, request):
         new_ip_address = get_real_ip(request)
         old_ip_address = request.session.get('ip_address', None)
 
+        reader = geoip2.database.Reader(settings.GEOIP_PATH)
         if not new_ip_address and old_ip_address:
             del request.session['ip_address']
             del request.session['country_code']
         elif new_ip_address != old_ip_address:
-            if new_ip_address.find(':') >= 0:
-                country_code = pygeoip.GeoIP(settings.GEOIPV6_PATH).country_code_by_addr(new_ip_address)
-            else:
-                country_code = pygeoip.GeoIP(settings.GEOIP_PATH).country_code_by_addr(new_ip_address)
+            response = reader.country(new_ip_address)
```

Comment:
same issue here in terms of dealing with geoip2.errors.AddressNotFoundError exception.